src/Bridge/Symfony/Bundle/Test/Constraint/MatchesJsonSchema.php

```diff
@@ -29,12 +30,16 @@ final class MatchesJsonSchema extends Constraint
     private $checkMode;
 
     /**
-     * @param array|string $schema
+     * @param Schema|array|string $schema
      */
     public function __construct($schema, ?int $checkMode = null)
     {
+        /** @var array|string $schema */
+        $schema = $schema instanceof Schema ? json_encode($schema) : $schema;
```

Comment:
```suggestion
        $schema = $schema instanceof \ArrayObject ? json_encode($schema) : $schema;
```

Reply:
Actually, what about not supporting this here at all?